# R6-Bionic: QEMU build integration for the Canon EOS R6 machine
#
# Apply inside a qemu-eos (QEMU 6.1 based) checkout, then copy r6.c:
#   git apply patches/qemu-r6.patch
#   cp patches/r6.c hw/arm/r6.c
#
# (The local qemu/ tree in this repo already has both changes applied.)

--- a/hw/arm/Kconfig
+++ b/hw/arm/Kconfig
@@ -40,6 +40,10 @@ config DIGIC
     bool
     select PTIMER
     select PFLASH_CFI02

+config CANON_R6
+    bool
+
 config EXYNOS4
     bool
     select A9MPCORE

--- a/configs/devices/arm-softmmu/default.mak
+++ b/configs/devices/arm-softmmu/default.mak
@@ -33,6 +33,7 @@
 CONFIG_A9SCU=y
 CONFIG_A9_GTIMER=y
 CONFIG_ARM_MPTIMER=y
+CONFIG_CANON_R6=y
 CONFIG_DIGIC=y

--- a/hw/arm/meson.build
+++ b/hw/arm/meson.build
@@ -5,6 +5,7 @@ arm_ss.add(files('boot.c'), fdt)
 arm_ss.add(when: 'CONFIG_PLATFORM_BUS', if_true: files('sysbus-fdt.c'))
 arm_ss.add(when: 'CONFIG_ARM_VIRT', if_true: files('virt.c'))
 arm_ss.add(when: 'CONFIG_ACPI', if_true: files('virt-acpi-build.c'))
 arm_ss.add(when: 'CONFIG_DIGIC', if_true: files('digic_boards.c'))
+arm_ss.add(when: 'CONFIG_CANON_R6', if_true: files('r6.c'))
 arm_ss.add(when: 'CONFIG_EXYNOS4', if_true: files('exynos4_boards.c'))
 arm_ss.add(when: 'CONFIG_EMCRAFT_SF2', if_true: files('msf2-som.c'))
 arm_ss.add(when: 'CONFIG_HIGHBANK', if_true: files('highbank.c'))
